web/src/app/chat/components/input/ActionManagement.tsx

```diff
@@ -439,79 +481,255 @@ function MCPToolsList({
+                        : "bg-neutral-300 dark:bg-neutral-700"
+                    }`}
+                    style={{
+                      width: "28px",
+                      height: "16px",
+                      borderRadius: "var(--Radius-Round, 1000px)",
+                    }}
+                    aria-pressed={isEnabled}
+                    aria-label={`Toggle ${tool.display_name || tool.name}`}
+                  >
+                    <div
+                      className={`absolute top-[2px] left-[2px] h-[12px] w-[12px] rounded-full transition-transform duration-200 ease-in-out ${
+                        isEnabled
+                          ? "translate-x-[12px] bg-white"
+                          : "translate-x-0 bg-white dark:bg-neutral-900"
+                      }`}
+                      style={{
+                        boxShadow: "0 0 1px 1px rgba(0, 0, 0, 0.05)",
+                      }}
+                    />
+                  </button>
+                </div>
+              );
+            })
+          )}
+        </div>
+        <div
+          className="sticky w-full pointer-events-none transition-opacity ease-out bg-gradient-to-t from-white to-transparent dark:from-neutral-900"
+          style={{
+            bottom: showReauthRow ? "40px" : "0px",
+            height: "24px",
+            opacity: showFadeMask ? 1 : 0,
+          }}
+        />
+        {showReauthRow && onReauthenticate && (
+          <div className="sticky bottom-0 bg-background-neutral-00 border-t border-border z-[1] rounded-b-lg">
+            <button
+              type="button"
+              onClick={onReauthenticate}
+              className="
+                w-full
+                flex
+                items-center
+                justify-between
+                px-2
+                py-2.5
+                text-left
+                bg-background-neutral-00
+                hover:bg-neutral-100
+                dark:hover:bg-neutral-800
+                rounded-b-lg
+                hover:rounded-b-lg
+                transition-colors
+              "
+            >
+              <div className="flex items-center gap-2">
+                {isReauthLoading ? (
+                  <FiLoader
+                    className="animate-spin text-text-400 dark:text-neutral-500"
+                    size={14}
+                  />
+                ) : (
+                  <FiKey
+                    className="text-text-500 dark:text-neutral-300"
+                    size={14}
+                  />
+                )}
+                <span className="text-sm font-medium text-neutral-700 dark:text-neutral-300">
+                  Re-Authenticate
+                </span>
+              </div>
+              {!isReauthLoading && (
+                <FiChevronRight
+                  size={14}
+                  className="text-neutral-400 dark:text-neutral-500 transition-colors"
+                />
+              )}
```

Comment:
Let's use the icons in the `web/src/icons` directory instead! There should be an `SvgChevronRight`. @Weves is bringing in the `SvgKey`, so you can wait until his PR lands to use it. The `FiLoader` should be replaced with the `SvgRefreshCw`, I believe.